fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/PartitionManager.java

```diff
@@ -0,0 +1,461 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.alibaba.fluss.server.coordinator;
+
+import com.alibaba.fluss.annotation.VisibleForTesting;
+import com.alibaba.fluss.cluster.MetadataCache;
+import com.alibaba.fluss.config.AutoPartitionTimeUnit;
+import com.alibaba.fluss.config.ConfigOptions;
+import com.alibaba.fluss.config.Configuration;
+import com.alibaba.fluss.exception.PartitionAlreadyExistsException;
+import com.alibaba.fluss.exception.PartitionNotExistException;
+import com.alibaba.fluss.exception.PartitionSpecInvalidException;
+import com.alibaba.fluss.metadata.PartitionSpec;
+import com.alibaba.fluss.metadata.TableInfo;
+import com.alibaba.fluss.metadata.TablePath;
+import com.alibaba.fluss.server.utils.TableAssignmentUtils;
+import com.alibaba.fluss.server.zk.ZooKeeperClient;
+import com.alibaba.fluss.server.zk.data.BucketAssignment;
+import com.alibaba.fluss.server.zk.data.PartitionAssignment;
+import com.alibaba.fluss.utils.AutoPartitionStrategy;
+import com.alibaba.fluss.utils.PartitionUtils;
+import com.alibaba.fluss.utils.clock.Clock;
+import com.alibaba.fluss.utils.clock.SystemClock;
+import com.alibaba.fluss.utils.concurrent.ExecutorThreadFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NavigableSet;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static com.alibaba.fluss.utils.PartitionUtils.generateAutoPartitionName;
+import static com.alibaba.fluss.utils.Preconditions.checkNotNull;
+import static com.alibaba.fluss.utils.concurrent.LockUtils.inLock;
+
+/**
+ * A partition manager which will manage static partition tables and auto partition tables.
+ *
+ * <p>For static partition tables, it's responsible for adding partitions to or drop partitions from
+ * static partition table.
+ *
+ * <p>For auto partition tables, it's responsible not only for adding partitions to or drop
+ * partitions from auto partition table, but also triggering auto partition for these auto partition
+ * tables in cluster periodically. It'll use a {@link ScheduledExecutorService} to schedule the auto
+ * partition which trigger auto partition for them.
+ */
+public class PartitionManager implements AutoCloseable {
```

Comment:
I don't think we need such big refactoring for the `AutoPartitionManager`. The `addPartition` and `dropPartition` are simple metadata operations on zookeeper (should be supported on `MetadataManager` just like `create/dropTable`). If it is a auto-partition table, `TableChangeWatcher` watches the changes of partition creation and deletion and pushes the events to   `AutoPartitionManager`, so no changes to `AutoPartitionManager`. If it is not a auto-partition table, then no changes to `AutoPartitionManager.

Reply:
Done!